cmd/endpoint.go

```diff
@@ -229,6 +227,14 @@ func NewEndpointList(args ...string) (endpoints EndpointList, err error) {
 
 	sort.Sort(endpoints)
 
+	if len(endpoints) > 1 {
+		// Check whether no. of args are valid for XL distribution.
+		if !isValidDistribution(len(endpoints)) {
```

Comment:
This check should be executed first to avoid unnecessary validation/creation of endpoints.

Reply:
This code is not new @balamurugana this is how it is in current master. But in any case i would suggest not review endpoint.go for now since it has to handle a new ellipses syntax and things will change.

Reply:
Unnecessary changes always confuse.  There is no point in making such changes which makes reviewing harder. 

Anyway this PR can't be reviewed without a prior discussion about design and implementation.

Reply:
I am very well aware of such things @balamurugana. I had to make those changes for demo purposes.

In any case i would like to point here constructive comments are more helpful and i don't see you are being helpful here neither your comments sound encouraging. 

FWIW design is finalized already with @abperiasamy . So go through the gist I shared feel free to add your comments  - https://gist.github.com/harshavardhana/72ff3680bd48f7f2d6c67cc45cae1a85

Reply:
I am very well aware of such things @balamurugana. I had to make those changes for demo purposes.

FWIW design is finalized already with @abperiasamy . So go through the gist I shared feel free to add your comments  - https://gist.github.com/harshavardhana/72ff3680bd48f7f2d6c67cc45cae1a85